server/src/main/scala/endpoints4s/uzhttp/server/Endpoints.scala

```diff
@@ -83,4 +83,15 @@ trait EndpointsWithCustomErrors extends algebra.EndpointsWithCustomErrors with R
    */
   def handleServerError(throwable: Throwable): RIO[Blocking, UzResponse] =
     serverErrorResponse(throwableToServerError(throwable))
+
+  override def choiceRequestEntity[A, B](
+    requestEntityA: RequestEntity[A],
+    requestEntityB: RequestEntity[B]
+  ): RequestEntity[Either[A, B]] =
+    req =>
+      requestEntityA(req).zip(requestEntityB(req)).map {
+        case (Valid(v), _)        => Valid(Left[A, B](v))
+        case (_, Valid(v))        => Valid(Right[A, B](v))
+        case (invalidA, invalidB) => invalidA.zip(invalidB).asInstanceOf[Invalid]
+      }
```

Comment:
It is correct implementation, I'm wondering if we can avoid extracting `B` when extracting `A` succeeds.
I don't see tests for it so that might be a reason why there are no issues when it is unimplemented.
Can you add test for it?